### ISSUE TITLE
gitgraph-react: truly reactive rendering without children prop callback

### ISSUE THREAD

**Issue by haizz** [nicoespeon/gitgraph.js#288]
Current React renderer takes a callback as children prop which provides `GitgraphUserApi` object which is supposed to be used to build a graph.

But, in cases when the source of graph data comes as an external prop (for example, passed as GraphQL query result), it is very awkward to re-render the graph when source changes: callback/effect hooks and/or local state have to be involved, everything becomes quite messy. 

Plus it is currently impossible to ditch `GitgraphUserApi` and build `GitgraphCore` manually (in cases when `import` gives non desirable results, for example).

It would be great to be able to build the graph using `GitgraphUserApi` or `GitgraphCore` or whatever and then pass it to React renderer as plain simple prop. For example:

```typescript
function MyGitGraph(props: MyGitGraphProps) {
   const graph: GitgraphCore<SVGElement> = useMemo(() => {
      // build graph somehow by blahblah.import(props.gitJson)
   }, [props.gitJson]);
   return <Gitgraph graph={graph}/>;
}
```

**Comment by haizz** [comment]
I actually tried to implement re-rendering using React hooks, but ran into a problem with original Gitgraph React renderer not resetting its internal state, and since I decided to write my custom renderer anyway, I stopped looking into that and deleted all the code.

**Comment by nicoespeon** [comment]
Hey @haizz 👋 

This has been implemented thanks to @evaliyev in #306 and released in gitgraph@react v1.5.0.

Here's a Story to illustrate:

https://github.com/nicoespeon/gitgraph.js/blob/57cd512b0f170191f332eeda081e90feb005b8a1/packages/stories/src/gitgraph-react/1-basic-usage.stories.tsx#L21-L37
